### DISCUSSION TITLE
[mini.operators] `gs` mapping sorts words, can it also sort lines?

### DISCUSSION THREAD

**Discussion by ivuorinen** [nvim-mini/mini.nvim#1401] [General]
One of the most common tasks I find myself repeating is sorting lines alphabetically.

Should I just stick with using visual mode to select region and hitting the good old `:'<,'>sort`?

Is this something people should just set as a keymap and isn't even worth considering part of the plugins functionality?

If it's not worth expanding the current functionality (or if I missed it somehow, sorry!) could [the wording on the help page](https://github.com/echasnovski/mini.nvim/blob/7ebfab26d77a4b9b05aaae565907e7fa4b2ee154/doc/mini-operators.txt#L210) be changed to something like `Sort words (does not sort lines by default)`?

(Thank you for creating such an amazing set of time saving plugins!)

**Comment by echasnovski** [comment]
Yes, `gs` can sort lines. It is indeed one of the most common use cases for it, after all :)

Sorting [by default uses `default_sort_func()`](https://github.com/echasnovski/mini.nvim/blob/7ebfab26d77a4b9b05aaae565907e7fa4b2ee154/doc/mini-operators.txt#L271-L272), which in turns decides how to sort [based on the mode](https://github.com/echasnovski/mini.nvim/blob/7ebfab26d77a4b9b05aaae565907e7fa4b2ee154/doc/mini-operators.txt#L437-L446).

For example:
- `gsi)` will sort arguments inside parenthesis. Because `i)` is charwise textobject. Same can be achieved with `vi)gs` (`v` starts charwise Visual mode).
- `gsip` will sort lines inside paragraph. Because `ip` is linewise textobject. Same can be achieved with `Vipgs` (`V` starts linewise Visual mode) and also `vipgs` by accident (because "When used in Visual mode it is made linewise.", see `:h ip`).

**Reply by ivuorinen** [reply]
Thank you so much for your amazing answer! More than I could have hoped for!
